src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -53,6 +56,16 @@ protected JSRuntime()
         /// </summary>
         protected TimeSpan? DefaultAsyncTimeout { get; set; }
 
+        /// <summary>
+        /// Contains the combined byte array(s) being serialized.
+        /// </summary>
+        internal readonly List<byte[]> ByteArraysToSerialize = new();
```

Comment:
We need to rethink this a bit.

The problem here is that if I serialize 10K byte arrays this list will grow up to 10K buckets and won't shrink when you call `Clear` afterwards.

It might be better to implement a custom type for this, and use an array pool internally to track the byte arrays.

Reply:
The `ArrayBuilder<T>` class in the `Shared` sources might be a convenient fit. It grows by renting from an `ArrayPool` and when you call `Clear` it returns the current buffer to the pool. So the pooling concerns would be handled implicitly.

Reply:
Thanks, https://github.com/dotnet/aspnetcore/pull/32259/commits/cdba736891484cc32bd66e49646d0156656e547d should handle this 😄